modules/compatibility/src/test/java/org/apache/ignite/compatibility/testframework/junits/IgniteCompatibilityAbstractTest.java

```diff
@@ -0,0 +1,331 @@
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.resource.GridSpringResourceContext;
+import org.apache.ignite.internal.util.lang.GridAbsPredicate;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.testframework.junits.multijvm.IgniteProcessProxy;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Super class for all compatibility tests.
+ */
+public abstract class IgniteCompatibilityAbstractTest extends GridCommonAbstractTest {
+    /** Using for synchronization of nodes startup in case of starting remote nodes first. */
+    public static final String SYNCHRONIZATION_LOG_MESSAGE_JOINED = "Remote node has joined, id: ";
+
+    /** Using for synchronization of nodes startup in case of starting remote nodes first. */
+    public static final String SYNCHRONIZATION_LOG_MESSAGE_PREPARED = "Remote node has prepared, id: ";
+
+    /** Waiting seconds of the join of a node to topology. */
+    protected static final int NODE_JOIN_TIMEOUT = 30;
+
+    /** Local JVM Ignite node. */
+    protected Ignite locJvmInstance = null;
+
+    /** Remote JVM Ignite instance. */
+    protected Ignite rmJvmInstance = null;
+
+    /**
+     * Gets a path to the default DB working directory.
+     *
+     * @return Path to the default DB working directory.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #deleteDefaultDBWorkDirectory()
+     * @see #defaultDBWorkDirectoryIsEmpty()
+     */
+    protected Path getDefaultDbWorkPath() throws IgniteCheckedException {
+        return Paths.get(U.defaultWorkDirectory() + File.separator + "db");
+    }
+
+    /**
+     * Deletes the default DB working directory with all sub-directories and files.
+     *
+     * @return {@code true} if and only if the file or directory is successfully deleted, otherwise {@code false}.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #getDefaultDbWorkPath()
+     * @see #deleteDefaultDBWorkDirectory()
+     */
+    protected boolean deleteDefaultDBWorkDirectory() throws IgniteCheckedException {
+        Path dir = getDefaultDbWorkPath();
+
+        return Files.notExists(dir) || U.delete(dir.toFile());
+    }
+
+    /**
+     * Checks if the default DB working directory is empty.
+     *
+     * @return {@code true} if the default DB working directory is empty or doesn't exist, otherwise {@code false}.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #getDefaultDbWorkPath()
+     * @see #deleteDefaultDBWorkDirectory()
+     */
+    @SuppressWarnings("ConstantConditions")
+    protected boolean defaultDBWorkDirectoryIsEmpty() throws IgniteCheckedException {
```

Comment:
I think method name should be `isDefaultDBWorkDirectoryEmpty`

Reply:
Agree, will be fixed.